enhancements/sdk-external-and-pluggable-validations.md

```diff
@@ -0,0 +1,494 @@
+---
+title: sdk-external-and-pluggable-validations
+authors:
+  - "@jmrodri"
+reviewers:
+  - "@camilamacedo86"
+  - "@joelanford"
+  - "@bparees"
+approvers:
+  - "@camilamacedo86"
+  - "@joelanford"
+creation-date: 2021-10-01
+last-updated: 2021-10-01
+status: implementable
+see-also:
+  - "/enhancements/this-other-neat-thing.md"
+---
+
+# SDK External and Pluggable Validations
+
+## Release Signoff Checklist
+
+- [X] Enhancement is `implementable`
+- [X] Design details are appropriately documented from clear requirements
+- [ ] Test plan is defined
+- [ ] Graduation criteria for dev preview, tech preview, GA
+
+## Open Questions [optional]
+
+ 1. what validations do we have today?
+ 1. can we use scorecard to replace all of these validations?
+    * scorecard uses the cluster to run the tests, these validations typically
+      run locally or in a pipeline. They are also done before the expensive
+      operator tests are run.
+ 1. what would it take to convert an existing validator to an executable format?
+
+## Summary
+
+Today, validations used by CVP are compiled into the operator-sdk. Any changes
+to the validation rules requires a release of operator-framework/api followed by
+a release of operator-sdk. This proposal attempts to design a way where the
+validations can be hosted in their own repos as well as updated without
+requiring new releases of the operator-sdk.
+
+## Motivation
+
+Every time the business changes validation rules, it requires an update to the
+operator-framework/api library. A release of said library, then it needs to get
+included into the operator-sdk. Then a release of the SDK needs to be cut in
+order for the new validation rule change to be usable.
+
+This process slows down the business by having to wait for this release process
+to occur for what could be a very small rule change. It also causes downstream
+rules to require an immediate upstream operator-sdk release, which may
+otherwise be as far as 3 weeks away. It is difficult to explain to the community
+why we need a new release for a CVP need.
+
```

Comment:
WDYT about add here as motivation:
- Able to have vendor-like validators and custom validators created by the users?

It seems for me the strongest motivation since today we cannot ship vendor-like ones in the operator-framework/api.

Reply:
@camilamacedo86 what do you mean by "vendor-like"?

Reply:
Checks specifics for OCP k8s vendor for e.g.

Reply:
> Checks specifics for OCP k8s vendor for e.g.

OH! vendor *specific* validations. I will add that to the list. Thanks for clarifying.